frontend/src/store/bucketStore.ts

```diff
@@ -50,31 +51,39 @@ export const useBucketStore = defineStore('bucket', () => {
 
       const searchPerms = (await bucketService.searchForPermissions({ bucketId })).data;
 
-      // TODO: Feed a comma separated list instead of searching individual users once COMS accepts that
-      const uniqueIds = [...new Set(searchPerms.map((x: any) => x.userId))]; //.join(',');
-      const uniqueUsers: any = [];
-
-      await Promise.all(
-        uniqueIds.map(async (x: any) => {
-          const userResponse = await userService.searchForUsers({ userId: x });
-          uniqueUsers.push(userResponse.data[0]);
-        })
-      );
-
-      const userPermissions: UserPermission[] = [];
-      uniqueUsers.forEach((user: any) => {
-        userPermissions.push({
-          userId: user.userId,
-          fullName: user.fullName,
-          create: searchPerms.some((perm: any) => perm.userId === user.userId && perm.permCode === Permissions.CREATE),
-          read: searchPerms.some((perm: any) => perm.userId === user.userId && perm.permCode === Permissions.READ),
-          update: searchPerms.some((perm: any) => perm.userId === user.userId && perm.permCode === Permissions.UPDATE),
-          delete: searchPerms.some((perm: any) => perm.userId === user.userId && perm.permCode === Permissions.DELETE),
-          manage: searchPerms.some((perm: any) => perm.userId === user.userId && perm.permCode === Permissions.MANAGE),
+      if (searchPerms[0]) {
+        const perms = searchPerms[0].permissions;
+
+        // TODO: Feed a comma separated list instead of searching individual users once COMS accepts that
+        // const uniqueIds = [...new Set(perms.map((x: any) => x.userId))].join(',');
+        // const uniqueUsers = await userService.searchForUsers({ userId: uniqueIds });
+
+        const uniqueIds = [...new Set(perms.map((x: any) => x.userId))];
+        const uniqueUsers: any = [];
+
+        await Promise.all(
+          uniqueIds.map(async (x: any) => {
+            const userResponse = await userService.searchForUsers({ userId: x });
+            uniqueUsers.push(userResponse.data[0]);
+          })
+        );
+
+        const userPermissions: UserPermission[] = [];
+        uniqueUsers.forEach((user: any) => {
+          userPermissions.push({
+            userId: user.userId,
+            elevatedRights: config.value.idpList.find((idp: any) => idp.idp === user.idp)?.elevatedRights ?? false,
```

Comment:
side note: as this expression might get reused in a few places might be worth shortening to: `config.value.idpList.find((idp: any) => idp.idp === user.idp).elevatedRights`

Reply:
We should also be avoiding the type of any if possible - that means we probably want to be defining this object type in our interfaces and pulling that in.